### PR TITLE
Update CLI developer guide

### PR DESCRIPTION
**Pull request by piotrpMSFT**

### CODE REVIEW

**Comment by piotrpMSFT** [comment]
Fixes #4347

**Comment by piotrpMSFT** [comment]
@mishra14 I've updated the developer guide to match its previous intent, though this really meant removing a bunch of content that no longer applies. I expect there are other contents that you'd like to see, so please do comment with any requests.

File: Documentation/developer-guide.md [hunk 1]
```diff
@@ -30,61 +27,48 @@ In order to build .NET Command Line Interface, you need the following installed
 
 ## Building/Running
 
-1. Run `build.cmd` or `build.sh` from the root depending on your OS. If you don't want to execute tests, run `build.cmd -Targets Prepare,Compile` or `./build.sh --targets Prepare,Compile`.
-2. Use `artifacts/{os}-{arch}/stage2/dotnet` to try out the `dotnet` command. You can also add `artifacts/{os}-{arch}/stage2` to the PATH if you want to run `dotnet` from anywhere.
+1. Run `build.cmd` or `build.sh` from the root depending on your OS. If you don't want to execute tests, run `build.cmd /t:Compile` or `./build.sh /t:Compile`.
```
Comment (livarcocc) [inline Documentation/developer-guide.md]: NIT: Remove the extra space at the end of the line.

File: Documentation/developer-guide.md [hunk 2]
```diff
@@ -30,61 +27,48 @@ In order to build .NET Command Line Interface, you need the following installed
 
 ## Building/Running
 
-1. Run `build.cmd` or `build.sh` from the root depending on your OS. If you don't want to execute tests, run `build.cmd -Targets Prepare,Compile` or `./build.sh --targets Prepare,Compile`.
-2. Use `artifacts/{os}-{arch}/stage2/dotnet` to try out the `dotnet` command. You can also add `artifacts/{os}-{arch}/stage2` to the PATH if you want to run `dotnet` from anywhere.
+1. Run `build.cmd` or `build.sh` from the root depending on your OS. If you don't want to execute tests, run `build.cmd /t:Compile` or `./build.sh /t:Compile`. 
+2. Use `artifacts/{RID}/stage2/dotnet` to try out the `dotnet` command. You can also add `artifacts/{os}-{arch}/stage2` to the PATH if you want to run `dotnet` from anywhere.
```
Comment (livarcocc) [inline Documentation/developer-guide.md]: Should this say: if you want to run `dotnet` from everywhere?

File: Documentation/developer-guide.md [hunk 3]
```diff
@@ -30,61 +27,48 @@ In order to build .NET Command Line Interface, you need the following installed
...
+4. `dotnet run3`
 
 ## Running tests
 
-All the CLI tests are located under `test`. In order to run them, after doing a restore on the CLI repo just do the following:
-
-1. Navigate to a test project, for instance: `cd test\dotnet-test.UnitTests`
-2. `dotnet test`
+1. To run all tests invoke `build.cmd` or `build.sh` which will build the product and run the tests.
+2. To run a specific test, cd into that test's directory and execute `dotnet test`
```
Comment (livarcocc) [inline Documentation/developer-guide.md]: I think we should mention that for the CLI tests itself, it expects stage2 to be on the path and the NUGET_PACKAGES should be set to <repo_root>/.nuget/packages

File: Documentation/developer-guide.md (hunk 2 above)
Comment (brthor) [inline Documentation/developer-guide.md]: change the other `{os}-{arch}` to {RID} too

File: Documentation/developer-guide.md [hunk 4]
```diff
@@ -30,61 +27,48 @@ In order to build .NET Command Line Interface, you need the following installed
...
-2. Use `artifacts/{os}-{arch}/stage2/dotnet` to try out the `dotnet` command. You can also add `artifacts/{os}-{arch}/stage2` to the PATH if you want to run `dotnet` from anywhere.
+1. Run `build.cmd` or `build.sh` from the root depending on your OS. If you don't want to execute tests, run `build.cmd /t:Compile` or `./build.sh /t:Compile`. 
+2. Use `artifacts/{RID}/stage2/dotnet` to try out the `dotnet` command. You can also add `artifacts/{os}-{arch}/stage2` to the PATH if you want to run `dotnet` from anywhere.
 
 ## A simple test
 
-1. `cd TestAssets\TestProjects\TestSimpleIncrementalApp`
-2. `dotnet restore`
-3. `dotnet run`
+1. `cd {new directory}`
```
Comment (brthor) [inline Documentation/developer-guide.md]: Is this guide for dotnet devs or cli repo devs?

**Comment by mishra14** [comment]
@piotrpMSFT 👍
Looks good to me.

File: Documentation/developer-guide.md (hunk 2 above)
Comment (piotrpMSFT) [inline Documentation/developer-guide.md]: fixed.

File: Documentation/developer-guide.md (hunk 1 above)
Comment (piotrpMSFT) [inline Documentation/developer-guide.md]: fixed.

File: Documentation/developer-guide.md (hunk 4 above)
Comment (piotrpMSFT) [inline Documentation/developer-guide.md]: This guide is for developers who want to work in the CLI repo or extend the CLI in their own repos. I'm sure there is a bunch of missing content here, so please do suggest any additions you think we need. As @mishra14 pointed out in the source issue, this document has been neglected.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX x64 Release Build please.

**Comment by SpiraMira** [comment]
Hi - the docs have changed the build.sh command line syntax from the -- convention.  Yet the scripts build.sh and run-build.sh indicate otherwise. Confused.

Also, I need to build on Sierra (which currently breaks), how do i pass the /p:SkipInvalidConfigurations=true now and does that bypass the GetKernelRelease() check from PlatformAbstractions?

Thanks...

**Pull request merged**